oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -34,19 +35,25 @@ pub use channel::{Handle, HandleDirection};
 
 #[derive(Debug)]
 struct Node {
-    reference: NodeRef,
-    join_handle: JoinHandle<()>,
+    reference: NodeId,
+    join_handle: Option<JoinHandle<()>>,
 
     /// The Label associated with this node.
     ///
     /// This is set at node creation time and does not change after that.
     ///
     /// See https://github.com/project-oak/oak/blob/master/docs/concepts.md#labels
     label: oak_abi::label::Label,
+
+    handles: Mutex<HashSet<Handle>>,
 }
 
 #[derive(PartialEq, Eq, Hash, Clone, Copy, Debug)]
-pub struct NodeRef(u64);
+pub struct NodeId(u64);
```

Comment:
Could you add a comment to explain the desired properties / invariants of this type? In particular, the fact that node implementations should not be able to forge instances of this type.

Reply:
Done

Reply:
I don't really understand the forgery concern.  The `NodeId` isn't even accessible to Wasm Nodes, so there's no opportunity to forge, and any pseudo-Nodes are part of the TCB anyhow.  (Also, an incrementing counter doesn't seem very unforgeable)